pkg/imageset/updateinstaller.go

```diff
@@ -185,8 +187,26 @@ func (o *UpdateInstallerImageOptions) Run() (returnErr error) {
 	}
 	o.log.WithField("cliImage", cliImage).Info("cli image found")
 
+	releaseMetadataRaw, err := ioutil.ReadFile(filepath.Join(o.WorkDir, releaseMetadataFilename))
+	if err != nil {
+		return errors.Wrapf(err, "could not read %s file", releaseMetadataFilename)
+	}
+	releaseMetadata := &cincinnatiMetadata{}
+	if err := json.Unmarshal(releaseMetadataRaw, releaseMetadata); err != nil {
+		return errors.Wrap(err, "unable to load release release-metadata")
+	}
+	if releaseMetadata.Kind != "cincinnati-metadata-v0" {
+		return errors.Wrap(err, "unrecognized release-metadata in release payload")
+	}
+
+	releaseVersion := getReleaseVersion(releaseMetadata, is)
+	if releaseVersion == "" {
+		return errors.New("no release version set in the release payload")
+	}
```

Comment:
Are you aware of any times the format has changed for this data recently? Assuming the format is consistent with what we've done already, just wanted to check if you know of any instances where this would have broken in the past, or if we should plan for it to break in the future. (i.e. handle the error a little more conservatively so the job still succeeds)

Reply:
so this format hasn't changed from the start AFAIK. to test and make sure this has been the same i tested the files for 4.6, 4.5, 4.4, 4.3

```
➜  hive git:(cd_release_image_version) ✗ podman run -ti --rm --entrypoint bash quay.io/openshift-release-dev/ocp-release@sha256:3e855ad88f46ad1b7f56c312f078ca6adaba623c5d4b360143f9f82d2f349741
[root@e399cd6a9496 /]# cat /release-manifests/release-metadata
{
  "kind": "cincinnati-metadata-v0",
  "version": "4.6.16",
  "previous": [
    "4.5.27",
    "4.5.28",
    "4.5.29",
    "4.5.30",
    "4.5.31",
    "4.6.1",
    "4.6.12",
    "4.6.13",
    "4.6.15",
    "4.6.3",
    "4.6.4",
    "4.6.6",
    "4.6.8",
    "4.6.9"
  ],
  "metadata": {
    "description": "",
    "url": "https://access.redhat.com/errata/RHBA-2021:0308"
  }
}[root@e399cd6a9496 /]# exit
➜  hive git:(cd_release_image_version) ✗ podman run -ti --rm --entrypoint bash quay.io/openshift-release-dev/ocp-release@sha256:27951dd757d472bf913daaffa548b865e87968831ca6f42c1f6946f7dcf0564e
[root@26aad6f51a0a /]# cat /release-manifests/release-metadata
{
  "kind": "cincinnati-metadata-v0",
  "version": "4.5.31",
  "previous": [
    "4.4.31",
    "4.4.32",
    "4.4.33",
    "4.5.17",
    "4.5.18",
    "4.5.19",
    "4.5.20",
    "4.5.21",
    "4.5.22",
    "4.5.23",
    "4.5.24",
    "4.5.27",
    "4.5.28",
    "4.5.29",
    "4.5.30"
  ],
  "metadata": {
    "description": "",
    "url": "https://access.redhat.com/errata/RHSA-2021:0313"
  }
}[root@26aad6f51a0a /]# exit
➜  hive git:(cd_release_image_version) ✗ podman run -ti --rm --entrypoint bash quay.io/openshift-release-dev/ocp-release-nightly@sha256:6d02e83e08c9ff183f9e6197117bdd3ff6598d50357b7a66eddf003b0b59f815
[root@8b0ed21ece60 /]# cat /release-manifests/release-metadata
{
  "kind": "cincinnati-metadata-v0",
  "version": "4.4.0-0.nightly-2020-06-01-021027",
  "previous": []
}[root@8b0ed21ece60 /]# exit
➜  hive git:(cd_release_image_version) ✗ podman run -ti --rm --entrypoint bash quay.io/openshift-release-dev/ocp-release@sha256:9ff90174a170379e90a9ead6e0d8cf6f439004191f80762764a5ca3dbaab01dc
[root@dff4ea937cbb /]# cat /release-manifests/release-metadata
{
  "kind": "cincinnati-metadata-v0",
  "version": "4.3.40",
  "previous": [
    "4.2.34",
    "4.2.36",
    "4.3.0-0.hotfix-2020-09-30-133631",
    "4.3.23",
    "4.3.25",
    "4.3.26",
    "4.3.27",
    "4.3.28",
    "4.3.29",
    "4.3.31",
    "4.3.33",
    "4.3.35",
    "4.3.38"
  ],
  "metadata": {
    "description": "",
    "url": "https://access.redhat.com/errata/RHSA-2020:4264"
  }
}[root@dff4ea937cbb /]# exit

```

Reply:
:+1: